robot/web/web.go

```diff
@@ -622,6 +623,19 @@ func (svc *webService) makeStreamServer(ctx context.Context) (*StreamServer, err
 		if isVideo {
 			config.AudioEncoderFactory = nil
 
+			// set TargetFrameRate to the framerate of the video source if available
+			props, err := svc.videoSources[name].MediaProperties(ctx)
+			if err != nil {
+				svc.logger.Warnw("failed to get video source properties", "name", name, "error", err)
+			} else if props.FrameRate > 0.0 {
+				// round float up to nearest int
+				config.TargetFrameRate = int(math.Ceil(float64(props.FrameRate)))
+			}
+			// default to 60fps if the video source doesn't have a framerate
+			if config.TargetFrameRate == 0 {
+				config.TargetFrameRate = 60
```

Comment:
Why do we default to `60`? What happens if we just leave `TargetFrameRate` at `0`

Reply:
If `TargetFrameRate` is 0 gostream will default to the process ticker to `keyFrameInterval`.

Using 60 instead of 30 as default as it is better to error on the side of faster processing.